### PR TITLE
Chapter 2/Homework 3

### PR DESCRIPTION
**Pull request by Tailor-AI**
First commit for homework 3.

### CODE REVIEW

File: Src/BootCamp.Chapter/Computer/DesktopComputer.cs [hunk 1]
```diff
@@ -2,6 +2,15 @@
 {
     public class DesktopComputer
     {
+        public DesktopComputer()
+        {
+            _body = new Body();
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Computer/DesktopComputer.cs]: Desktop computer has the same parts both for mac and for windows.
Each factory should be making branded parts and INSTALL those into a new computer.

**Review by Almantask — CHANGES_REQUESTED** [review]

File: Src/BootCamp.Chapter/Computer/DesktopComputer.cs (hunk 1 above)
Comment (Tailor-AI) [reply Src/BootCamp.Chapter/Computer/DesktopComputer.cs]: I'm not sure what you mean by "making branded parts" but now I made new parts in each factory and "installed" them in the assemble function.

File: Src/BootCamp.Chapter/MacFactory.cs [hunk 2]
```diff
@@ -1,26 +1,27 @@
...
 namespace BootCamp.Chapter
 {
     public class MacFactory : DesktopComputer
     {
-        public MacFactory()
-        {
-            Assemble();
-            Console.WriteLine(CompletedAssembly());
-        }
+        Body macBody = new Body();
```
Comment (Almantask) [inline Src/BootCamp.Chapter/MacFactory.cs]: This means that all computers from this factory will have the same, shared Body, Ram, Cpu, etc. It looks the same, yes, but it's a DIFFERENT INSTANCE

File: Src/BootCamp.Chapter/MsFactory.cs [hunk 3]
```diff
@@ -1,26 +1,27 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Text;
-using BootCamp.Chapter.Computer;
+﻿using BootCamp.Chapter.Computer;
 
 namespace BootCamp.Chapter
 {
     public class MsFactory : DesktopComputer
```
Comment (Almantask) [inline Src/BootCamp.Chapter/MsFactory.cs]: What dos Ms factory have different compared to Win factory? NOthing. Please make sure that Ms factory creates MsRam, MsBody, MsGpu, etc. And Mac creates MacRam, MacCpu, MacBody, etc.

**Review by Almantask — CHANGES_REQUESTED** [review]

File: Src/BootCamp.Chapter/Computer/DesktopComputer.cs (hunk 1 above)
Comment (Almantask) [reply Src/BootCamp.Chapter/Computer/DesktopComputer.cs]: What dos Ms factory have different compared to Win factory? NOthing. Please make sure that Ms factory creates MsRam, MsBody, MsGpu, etc. And Mac creates MacRam, MacCpu, MacBody, etc.

Comment (Tailor-AI) [reply Src/BootCamp.Chapter/Computer/DesktopComputer.cs]: Added Ms parts and Mac parts and moved the creation of the parts under the assemble function.

File: Src/BootCamp.Chapter/MacFactory.cs (hunk 2 above)
Comment (Tailor-AI) [reply Src/BootCamp.Chapter/MacFactory.cs]: Ok. Now when I moved these down to the assemble function there should be one instance for every computer?

File: Src/BootCamp.Chapter/MsFactory.cs (hunk 3 above)
Comment (Tailor-AI) [reply Src/BootCamp.Chapter/MsFactory.cs]: I removed the inheritance of DesktopComputer because these are not related I think?

**Review by Almantask — APPROVED** [review]
Looks good now

**Pull request closed without merging**